admin/admin-service/src/main/java/io/mosip/admin/service/impl/ApplicantDetailServiceImpl.java

```diff
@@ -0,0 +1,144 @@
+import org.json.JSONException;
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.ResourceAccessException;
+
+import java.util.*;
+
+@Service
+public class ApplicantDetailServiceImpl implements ApplicantDetailService {
+
+    private static final String FACE = "Face";
+
+    @Autowired
+    RestClient restClient;
+
+    @Autowired
+    private RidValidator<String> ridValidator;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Autowired
+    private AuditUtil auditUtil;
+
+    @Autowired
+    CbeffToBiometricUtil cbeffToBiometricUtil;
+
+
+    @Autowired
+    Utility utility;
+
+    private static final String IDENTITY = "identity";
+
+    private static final String RESPONSE = "response";
+
+    private static final String DOCUMENTS="documents";
+
+    private static final String ApplicantPhoto = "applicantPhoto";
+
+    private static final String VALUE = "value";
+    private static final String DOB = "dob";
+
+    @Value("${mosip.admin.applicantDetails}")
+    private String[] applicantDetails;
+
+
+    @Override
+    public ApplicantDetailsDto getApplicantDetails(String rid) throws Exception {
+        ApplicantDetailsDto applicantDetailsDto =new ApplicantDetailsDto();
+        Map<String,String> applicantDataMap=new HashMap<>();
+        ConvertRequestDto convertRequestDto = new ConvertRequestDto();
+        List<String> pathsegments=new ArrayList<>();
+        String dateOfBirth=null;
+        String individualBiometrics=null;
+        pathsegments.add(rid);
+        String imageData=null;
+        try {
+			if (!ridValidator.validateId(rid)) {
+				throw new RequestException(ApplicantDetailErrorCode.RID_INVALID.getErrorCode(),
+                        ApplicantDetailErrorCode.RID_INVALID.getErrorMessage());
+			}
+            String response = restClient.getApi(ApiName.RETRIEVE_IDENTITY_API,pathsegments,"type","bio",String.class);
+            JSONObject responseObj= objectMapper.readValue(response,JSONObject.class);
+            if(response!=null && responseObj.containsKey("response")) {
+                JSONObject responseJsonObj=  utility.getJSONObject(responseObj,RESPONSE);
+                JSONObject identityObj=utility.getJSONObject(responseJsonObj,IDENTITY);
+                JSONArray documents=utility.getJSONArray(responseJsonObj,DOCUMENTS);
+                String idenitityJson=utility.getMappingJson();
+                JSONObject idenitityJsonObject=objectMapper.readValue(idenitityJson,JSONObject.class);
+                JSONObject mapperIdentity=utility.getJSONObject(idenitityJsonObject,IDENTITY);
+                List<String> mapperJsonKeys = new ArrayList<>(mapperIdentity.keySet());
+                for (String key: mapperJsonKeys){
```

Comment:
can we avoid this for loop and rather get the JSON value from for the valueObject and handle the null check, this is to avoid unwanted iterating

Reply:
taken care

Reply:
Resolved